.github/workflows/test.yml

```diff
@@ -9,6 +9,23 @@ on:
       - '.gitignore'
 
 jobs:
+  tests:
+    runs-on: ubuntu-latest
+    container:
+      image: python:3.12
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
```

Comment:
✅ Great use of unittest for running tests! Ensure that all critical paths are covered in your test cases.